app/src/ui/diff/text-diff.tsx

```diff
@@ -623,6 +623,7 @@ export class TextDiff extends React.Component<ITextDiffProps, ITextDiffState> {
 
   private getAndStoreCodeMirrorInstance = (cmh: CodeMirrorHost | null) => {
     this.codeMirror = cmh === null ? null : cmh.getEditor()
+    this.initDiffSyntaxMode()
```

Comment:
This is technically unrelated to the bug at hand but when debugging I noticed that `onSwapDoc` would trigger before we got the editor ref, making its call to `initDiffSyntaxMode` a noop.